### PR TITLE
Log / respect rate limits

### PR DESCRIPTION
**Pull request by zdrve**
Thank you for your contribution!

## Checklist
- [x] Verified against local install of kennel (using `path:` in Gemfile)
- [x] Added tests
- [ ] Updated Readme.md (if user facing behavior changed)

https://github.com/zendesk/kennel/actions/runs/8120685832/job/22198172379#step:7:27

### CODE REVIEW

File: lib/kennel/api.rb [hunk 1]
```diff
@@ -5,6 +5,8 @@ module Kennel
   class Api
     CACHE_FILE = ENV.fetch("KENNEL_API_CACHE_FILE", "tmp/cache/details")
 
+    RateLimitParams = Data.define(:limit, :period, :remaining, :reset, :name)
```
Comment (grosser) [inline lib/kennel/api.rb:8]: feels like you had too much golang for breakfast :D

**Review by grosser — APPROVED** [review]
nice 🎉

**Comment by grosser** [comment]
good to just merge right ... any gotchas you expect ?
... even if we do this in parallel it should be mostly fine right ?

**Comment by zdrve** [comment]
> good to just merge right ... any gotchas you expect ? ... even if we do this in parallel it should be mostly fine right ?

I think it's good, I've tested it out a fair bit (e.g. https://github.com/zendesk/kennel/actions/runs/8121811687 and others). 

If the rate limit hits, then each thread that hits the rate limit will sleep – so typically you'd suddenly see 10 messages (1 per thread) saying "Rate limited, sleeping for [e.g.] 25 seconds". Then they all wake up and try again.

**Comment by grosser** [comment]
v1.149.0

**Pull request merged**